crates/provider/src/shared/types.rs

```diff
@@ -0,0 +1,185 @@
+use std::{
+    collections::HashMap, fs::File, io::Write, os::unix::process::ExitStatusExt, path::PathBuf,
+    process::ExitStatus, vec,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum ZombieRole {
+    Temp,
+    Node,
+    BootNode,
+    Collator,
+    CumulusCollator,
+    Authority,
+    FullNode,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum PortName {
+    Prometheus,
+    Rpc,
+    RpcWs,
+    P2P,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ImagePullPolicy {
+    IfNotPresent,
+    Never,
+    Always,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FileMap {
+    local_file_path:  PathBuf,
+    remote_file_path: PathBuf,
+    unique:           bool,
```

Comment:
What does ```unique``` means here ?

Reply:
`unique` is a flag to identify if the file should be cached in the fileserver or not.

Reply:
Maybe we could replace it with something more explicit like ```is_cached``` ?

Reply:
We can use `is_unique`, since here if is unique we don't need to cache the file.